code/datums/components/crafting/crafting.dm

```diff
@@ -451,6 +451,15 @@
 	data["display_compact"] = display_compact
 
 	var/list/surroundings = get_surroundings(user)
+	var/fueled_welder_count = 0
+	for (var/obj/item/weldingtool/welder in surroundings["instances"][/obj/item/weldingtool])
+		fueled_welder_count++
+		if (welder.reagents.get_reagent_amount(/datum/reagent/fuel) == 0)
+			fueled_welder_count--
+
+	if(fueled_welder_count == 0)
+		surroundings["tool_behaviour"] -= "welder"
```

Comment:
Minor issue:

Not all `TOOL_WELDER`s are `/obj/item/weldingtool`s

The code as-is makes it impossible to use non-welding tool welding tools to weld (as the code above will always say "we have zero welding tools)

Reply:
I have updated the code and solved this issue, please have a look when you can.